transport/event_notifier.cc

```diff
@@ -38,53 +38,67 @@ void cql_server::event_notifier::unregister_connection(cql_server::connection* c
     _schema_change_listeners.erase(conn);
 }
 
+void cql_server::event_notifier::on_schema_change(std::function<void(cql_server::connection*)> notify, std::function<std::string(std::exception_ptr)> describe_error) {
+    auto listeners = boost::copy_range<std::vector<cql_server::connection*>>(_schema_change_listeners);
+    std::vector<cql_server::connection*> conns_to_shutdown;
+    conns_to_shutdown.reserve(_schema_change_listeners.size());
+    for (auto&& conn : listeners) {
+        if (conn->_pending_requests_gate.is_closed()) {
+            continue;
+        }
+        std::exception_ptr ex;
+        try {
+            notify(conn);
+        } catch (...) {
+            ex = std::current_exception();
+            elogger.warn("{}. Closing connection", describe_error(ex));
+            conns_to_shutdown.emplace_back(conn);
```

Comment:
Looking at `cql_server::connection::write_response`, there is nothing which can fail there. The writing happens in the background. What can fail is construction of the message (conn->make_schema_change_event()). But it is more likely that boost::copy_range() above fails, which is not handled. 

Maybe this effort to handle failure here is futile after all. Why not let the exception from the listener propagate up like from other listeners, and fail the schema change?

Reply:
Makes sense.  Initially I thought we need to squelch the errors to preserve the existing behavior but you convinced me otherwise.
I'll drop this change then.